rust/src/operations/write.rs

```diff
@@ -473,19 +473,59 @@ impl std::future::IntoFuture for WriteBuilder {
     }
 }
 
-fn can_cast_batch(from_schema: &ArrowSchema, to_schema: &ArrowSchema) -> bool {
-    if from_schema.fields.len() != to_schema.fields.len() {
+fn can_cast_batch_fields(from_fields: &Fields, to_fields: &Fields) -> bool {
+    if from_fields.len() != to_fields.len() {
         return false;
     }
-    from_schema.all_fields().iter().all(|f| {
-        if let Ok(target_field) = to_schema.field_with_name(f.name()) {
-            can_cast_types(f.data_type(), target_field.data_type())
+
+    from_fields.iter().all(|f| {
+        if let Some((_, target_field)) = to_fields.find(f.name()) {
+            if let (DataType::Struct(fields0), DataType::Struct(fields1)) =
+                (f.data_type(), target_field.data_type())
+            {
+                can_cast_batch_fields(fields0, fields1)
+            } else {
+                can_cast_types(f.data_type(), target_field.data_type())
+            }
         } else {
             false
         }
     })
 }
 
+fn can_cast_batch(from_schema: &ArrowSchema, to_schema: &ArrowSchema) -> bool {
+    can_cast_batch_fields(from_schema.fields(), to_schema.fields())
+}
```

Comment:
```suggestion
```
Since there is only one callsite of this method and it is not a public API, I figure it makes sense to just remove it

Reply:
I agree with you. I pushed new commit.
Thanks for the review.